app/charts/shared/use-width.tsx

```diff
@@ -1,46 +1,60 @@
-import { createContext, ReactNode, useContext } from "react";
+import React, { createContext, ReactNode, useContext } from "react";
 
+import { useTransitionStore } from "@/stores/transition";
 import { useResizeObserver } from "@/utils/use-resize-observer";
 
-export interface Margins {
+export type Margins = {
   top: number;
   right: number;
   bottom: number;
   left: number;
-}
-export type Width = number;
-export interface Bounds {
+};
+
+export type Bounds = {
   width: number;
   height: number;
   margins: Margins;
   chartWidth: number;
   chartHeight: number;
-}
+};
 
-const INITIAL_WIDTH: Width = 1;
+const INITIAL_WIDTH = 1;
 
 export const Observer = ({ children }: { children: ReactNode }) => {
-  const [resizeRef, width] = useResizeObserver<HTMLDivElement>();
+  const [ref, width] = useResizeObserver<HTMLDivElement>();
+  const setEnableTransition = useTransitionStore((state) => state.setEnable);
+
+  // Disable transitions during resize, so we don't have a "laggy" feeling
+  React.useEffect(() => {
+    setEnableTransition(false);
+    const timeout = setTimeout(() => {
+      setEnableTransition(true);
+    }, 500);
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [width, setEnableTransition]);
```

Comment:
Super nit: I think that maybe the effect is firing a bit too much ? Alternative way.


```
const usePrevious = <T,>(value: T, duration: number): T => {
  const [previousValue, setPreviousValue] = useState<T>(value);

  useEffect(() => {
    const timeoutId = setTimeout(() => {
      setPreviousValue(value)
    }, duration);

    return () => {
      clearTimeout(timeoutId);
    };
  }, [value, duration]);

  return previousValue;
};


const prev = usePrevious(width, 500)
const isResizing = prev !== width
useEffect(() => setEnableTransition(isResizing), [isResizing])
```

Reply:
Really like the idea, done, thanks 💯